go/service/usersearch.go

```diff
@@ -129,29 +130,34 @@ func (q *compiledQuery) scoreString(str string) (bool, float64) {
 
 var fieldsAndScores = []struct {
 	multiplier float64
+	plumb      bool // plumb the matched value to displayLabel
 	getter     func(*keybase1.ProcessedContact) string
 }{
-	{1.0, func(contact *keybase1.ProcessedContact) string { return contact.ContactName }},
-	{1.0, func(contact *keybase1.ProcessedContact) string { return contact.Component.ValueString() }},
-	{1.0, func(contact *keybase1.ProcessedContact) string { return contact.DisplayName }},
-	{0.8, func(contact *keybase1.ProcessedContact) string { return contact.DisplayLabel }},
-	{0.7, func(contact *keybase1.ProcessedContact) string { return contact.FullName }},
-	{0.7, func(contact *keybase1.ProcessedContact) string { return contact.Username }},
+	{1.5, true, func(contact *keybase1.ProcessedContact) string { return contact.ContactName }},
+	{1.0, true, func(contact *keybase1.ProcessedContact) string { return contact.Component.ValueString() }},
+	{1.0, false, func(contact *keybase1.ProcessedContact) string { return contact.DisplayName }},
+	{0.8, false, func(contact *keybase1.ProcessedContact) string { return contact.DisplayLabel }},
+	{0.7, false, func(contact *keybase1.ProcessedContact) string { return contact.FullName }},
+	{0.7, false, func(contact *keybase1.ProcessedContact) string { return contact.Username }},
```

Comment:
Where do these multiplier values come from? And why are they not all distinct?

Reply:
Arbitrarily chosen, we wanted to bump search results higher up in the list if query matches contact name or component value.